corehq/apps/sms/mixin.py

```diff
@@ -132,64 +128,42 @@ def validate_number_format(cls, phone_number):
         if (not phone_number) or (not phone_number_re.match(phone_number)):
             raise InvalidFormatException("Phone number format must consist of only digits.")
 
-    def verify_unique_number(self, phone_number):
-        """
-        Verifies that the given phone number is not already in use by any other contacts.
-
-        return  void
-        raises  InvalidFormatException if the phone number format is invalid
-        raises  PhoneNumberInUseException if the phone number is already in use by another contact
-        """
+    def _create_phone_entry(self, phone_number):
         from corehq.apps.sms.models import PhoneNumber
+        return PhoneNumber.objects.create(
+            domain=self.domain,
+            owner_doc_type=self.doc_type,
+            owner_id=self.get_id,
+            phone_number=phone_number,
+            verified=False,
+            pending_verification=False,
+            is_two_way=False
+        )
+
+    def create_phone_entry(self, phone_number):
+        phone_number = apply_leniency(phone_number)
         self.validate_number_format(phone_number)
-        v = PhoneNumber.by_phone(phone_number, include_pending=True)
-        if v is not None and (v.owner_doc_type != self.doc_type or v.owner_id != self.get_id):
-            raise PhoneNumberInUseException("Phone number is already in use.")
+        return self._create_phone_entry(phone_number)
 
-    def save_verified_number(self, domain, phone_number, verified, backend_id=None, ivr_backend_id=None, only_one_number_allowed=False):
-        """
-        Saves the given phone number as this contact's verified phone number.
+    def get_or_create_phone_entry(self, phone_number):
+        with CriticalSection([self.phone_sync_key]):
+            return self._get_or_create_phone_entry(phone_number)
 
-        backend_id - the name of an SMSBackend to use when sending SMS to
-            this number; if specified, this will override any project or
-            global settings for which backend will be used to send sms to
-            this number
+    def _get_or_create_phone_entry(self, phone_number):
+        phone_number = apply_leniency(phone_number)
+        self.validate_number_format(phone_number)
 
-        return  The PhoneNumber
-        raises  InvalidFormatException if the phone number format is invalid
-        raises  PhoneNumberInUseException if the phone number is already in use by another contact
-        """
-        from corehq.apps.sms.models import PhoneNumber
+        entries = self.get_phone_entries()
```

Comment:
yeah exactly, we call this in a few places in the framework, and we almost always will get back a cached result so this helps us avoid hitting the db